tap_salesforce/client.py

```diff
@@ -258,13 +259,10 @@ def _make_request(self, method, path, data=None, params=None) -> requests.Respon
         )
 
         if resp.status_code < 200 or resp.status_code > 299:
-            # Salesforce error json body looks like
-            # [{'message': 'Your query request was running for too long.', 'errorCode': 'QUERY_TIMEOUT'}]
-            err_array = resp.json()
-            if not isinstance(err_array, list) or not isinstance(err_array[0], dict):
-                resp.raise_for_status()
-            
-            raise SalesforceException(err_array[0]['message'], err_array[0]['errorCode'])
+            ex = build_salesforce_exception(resp)
+            if ex:
+                raise ex
```

Comment:
Ok I really dislike this. :sweat_smile: 

If this is your favored approach to it, consider something like

```python
if resp.status_code >= 400:
    raise build_salesforce_exception(resp.json())

# completely forget about resp.raise_for_status()
```

Reply:
My primary thing here was to question if you knew whether the "Salesforce error format" is present for response codes `>= 500`, or if it's different.

If it's the same, the above solves the entire thing nicely.

Reply:
which doesn't scale so good so the equal is what you dislike:

```
if resp.status_code >= 400: 
    ex = build_salesforce_exception(resp.json()) 
    if ex:
        raise ex
    raise build_my_http_exception(resp) # or resp.raise_for_status()
```